kernel/aster-nix/src/process/clone.rs

```diff
@@ -453,3 +475,12 @@ fn set_parent_and_group(parent: &Arc<Process>, child: &Arc<Process>) {
 
     process_table_mut.insert(child.pid(), child.clone());
 }
+
+pub fn unshare(unshare_flags: CloneFlags) -> Result<()> {
```

Comment:
I think this logic can be moved into the implementation of `sys_unshare` syscall.

Reply:
I think `clone_namespaces` used in unshare should not be exposed to syscall, because unshare may need to add more features in the future, such as support for `CLONE_FS`, `CLONE_FILES` and other flags.